packages/smithy-core/src/smithy_core/documents.py

```diff
@@ -633,3 +638,24 @@ def read_document(self, schema: "Schema") -> Document:
     @override
     def read_timestamp(self, schema: "Schema") -> datetime.datetime:
         return self._value.as_timestamp()
+
+
+# A registry for on-demand deserialization of types by using a mapping of shape IDs to their deserializers.
+class TypeRegistry:
+    def __init__(
+        self,
+        types: dict[ShapeID, type[DeserializeableShape]],
+        sub_registry: "TypeRegistry | None" = None,
+    ):
```

Comment:
```suggestion
    def __init__(
        self,
        types: Mapping[ShapeID, type[DeserializeableShape]],
        sub_registry: "TypeRegistry | None" = None,
    ):
        """Initialize a TypeRegistry.
        
        :param types: A mapping of ShapeID to the shapes they deserialize to.
        :param sub_registry: A registry to delegate to if an ID cannot be found in
            types.
        """
```

Or something like that